minor updates; stylistic touch-ups

enze-chen opened the pull request:
Content updates are as follows:
* Added `client.data.delete_dataset()` method.
* Added a couple more examples to SearchClient tutorial.
* Added a SL demo notebook using a steel fatigue dataset.

Also cleaned up the code to have better readability and flow based on iterative feedback and improvements from the past several months.

enze-chen committed:
minor updates; stylistic touch-ups

enze-chen committed:
fixed package error

maxhutch commented:
The diff here is extensive.  Could you comment on the line changes that are more than stylistic touch-ups?

enze-chen commented:
@maxhutch Hmm, 1) Perhaps I used the wrong language, and 2) Perhaps it's a little intimidating, but I actually do think the changes leading to the massive diff, besides the three points I named, are stylistic in nature. I can certainly comment on what those are:
- More direct import of Python packages, e.g. `from time import sleep` instead of `import time`
- Specifying the input keys and arguments for functions on newlines,  e.g. 
```python
CitrinationClient(api_key=key, 
                  site=site)
``` 
instead of `CitrinationClient(key, site)`.
- More consistent formatting on code snippets in Markdown blocks, with inline `backticks` instead of **bold**.
- Adding some descriptive texts around naked code blocks, particularly when they appear consecutively.
- Consistent language / names.
- Displaying more URLs to show how API corresponds to the web interface.
- Removed redundant, dead code. Reworded awkward parts.
- Simplified docstrings and made them more consistent.

enze-chen commented on citrination_api_examples/clients_sequence/1_data_client_api_tutorial.ipynb:
```diff
@@ -353,10 +379,28 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# data_client.delete_dataset(dataset_id=dataset_id)"
```
Added a block for `client.data.delete_dataset()`. It's commented out so someone doesn't accidentally do this.

enze-chen commented on citrination_api_examples/clients_sequence/4_search_client_api_tutorial.ipynb:
```diff
@@ -225,6 +229,73 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Example: Filter a range of values\n",
```
added an example to create a `Filter` with `min` and `max`

enze-chen commented on citrination_api_examples/clients_sequence/4_search_client_api_tutorial.ipynb:
```diff
@@ -225,6 +229,73 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Example: Logic\n",
```
Added an example with a `Filter` with `logic`.

enze-chen commented on citrination_api_examples/clients_sequence/6_sequential_learning_steel_fatigue.ipynb:
```diff
@@ -0,0 +1,470 @@
+   "source": [
+    "<div class = \"intro\">\n",
+    "\n",
+    "# Sequential Learning Workshop\n",
```
This SL notebook uses a steel fatigue dataset instead of a toy function/data.

enze-chen commented on citrination_api_examples/clients_sequence/steel_fatigue_wrapper_class.py:
```diff
@@ -0,0 +1,706 @@
+'''
+Authors: Eddie Kim, Enze Chen, Nils Persson
```
This is the corresponding wrapper for the steel fatigue SL demo.

enze-chen commented:
@maxhutch Added comments where the changes actually took place. Thanks. :)

maxhutch reviewed (changes requested):
The changes look fine, but there is some weird formatting in the github render of:
https://github.com/CitrineInformatics/learn-citrination/blob/update-api-tutorials/citrination_api_examples/clients_sequence/6_sequential_learning_steel_fatigue.ipynb

Do you know what's going on there?

enze-chen committed:
removed HTML formatting

enze-chen commented:
> Do you know what's going on there?   

Wow; good catch! That's some leftover HTML formatting that's hidden when the Markdown renders in Jupyter, but apparently not on GitHub! :O I removed all of it—*I think*.

enze-chen commented:
Hmm....it looks like there's a library dependency on [`ase`](https://gitlab.com/ase/ase) that suddenly requires Python 3.5+. It's when we try to import `dfttopif`. How should we proceed @maxhutch? Remove Python 2.7 from the support as well?
